hlabsorg/league-gg
# Lol 21 integrate riot api into application

## JIRA ticket(s)


https://hlabs.atlassian.net/browse/LOL-21

## Type of Change




## Changes


- add's .env and .env.example for riot api key
- add's `swr` package to make cached requests from browser
- creates api endpoint for summoners
- creates riot utility functions like  `riotRequest` to easily make authenticated requests to riot api
- creates functions to get riot account and get summoner account 
- creates `fetcher` function for `swr`
- creates `useSummoners` hook to fetch summoner from our api
- creates constants file for regions

## Screenshots


<img width="1245" alt="Screenshot 2024-11-21 at 7 49 45 PM" src="https://github.com/user-attachments/assets/4602dbe8-35ef-43e3-8db4-75b95106dd18">


## Tests


1.  run app locally
2. see that ssj4goget84 profile is loaded
